samcli/commands/build/build_context.py

```diff
@@ -25,20 +26,23 @@ class BuildContext:
 
     def __init__(
         self,
-        resource_identifier,
-        template_file,
-        base_dir,
-        build_dir,
-        cache_dir,
-        cached,
-        mode,
-        manifest_path=None,
-        clean=False,
-        use_container=False,
-        parameter_overrides=None,
-        docker_network=None,
-        skip_pull_image=False,
-    ):
+        resource_identifier: Optional[str],
+        template_file: str,
+        base_dir: Optional[str],
+        build_dir: str,
+        cache_dir: str,
+        cached: bool,
+        mode: Optional[str],
+        manifest_path: Optional[str] = None,
+        clean: bool = False,
+        use_container: bool = False,
+        # pylint: disable=fixme
+        # FIXME: parameter_overrides is never None, we should change this to "dict" from Optional[dict]
+        # See samcli/commands/_utils/options.py:251 for its all possible values
+        parameter_overrides: Optional[dict] = None,
```

Comment:
what is the reason for not applying the fix now?

Reply:
Oh we want to keep this PR small and make the code review easier. This PR only focuses on adding types without behavior change. So we can merge it as quick as possible. I will create a follow up for those fixmes.